app/plugin_api.go

```diff
@@ -682,8 +682,11 @@ func (api *PluginAPI) GetPost(postID string) (*model.Post, *model.AppError) {
 	return post, appErr
 }
 
-func (api *PluginAPI) GetPostsSince(channelID string, time int64) (*model.PostList, *model.AppError) {
-	list, appErr := api.app.GetPostsSince(model.GetPostsSinceOptions{ChannelId: channelID, Time: time})
+func (api *PluginAPI) GetPostsSince(channelID string, time int64, page, perPage int) (*model.PostList, *model.AppError) {
```

Comment:
At the RPC level, this should be fine, but semantically, I think it's still a breaking change if old plugins are expecting to get all the posts from a call to `GetPostsSince`.

Might we consider handling that by treating `page == perPage == 0` specially?

At the same time, I'm also not 5/5 for this. It /feels/ unlikely that a plugin was depending on this behaviour, so we might be able to at least scan the known plugins for usage and make a judgement call here instead.

Reply:
> old plugins are expecting to get all the posts

Even before it wasn't returning all the posts _technically_, since it always had a limit. It's just that, the limit was quite big.

> scan the known plugins for usage

Any guidance on this, how or where can I verify it?

> handling that by treating page == perPage == 0

You mean, don't limit/paginate in this case?

Reply:
> technically, since it always had a limit. It's just that, the limit was quite big.

@vish9812, for reference, what was that big limit?

> Any guidance on this, how or where can I verify it?

@mickmister / @levb, I know we've done analysis of this kind in the past. (I sometimes do it locally just be grepping over my `plugins` src directories to see if anything matches.) Also interested in your feedback on maybe just ignoring it instead (allowing the "breaking change").

> You mean, don't limit/paginate in this case?

Yeah, the idea would be to essentially revert to the big limit as previously. Not sure it's worth this, as noted above.

Reply:
Sorry for the late response, grepping the local plugin repos is all I know of as far as analysis goes. 

As to the breaking change, this is an interface change and will break compatibility with the old plugin's signatures, no? I am not sure if the extra parameter is a breaking change for the RPC or not. Was introducing a new plugin API, `GetPostsSincePaginated` considered while keeping the old one intact, perhaps as deprecated?

Reply:
> As to the breaking change, this is an interface change and will break compatibility with the old plugin's signatures, no? I am not sure if the extra parameter is a breaking change for the RPC or not.

@levb @lieut-data IIRC this is a breaking change when the argument types of the function change. The "safe" way to introduce an API change is to add fields to an existing struct being used, but that's not an option here.

> Was introducing a new plugin API, `GetPostsSincePaginated` considered while keeping the old one intact, perhaps as deprecated?

+1 for introducing a new method. 3/5 An "ugly" method name is worth avoiding backwards compatibility issues

Reply:
It looks like the change was reverted for plugins though. Makes sense as it seems this PR is specifically meant for the webapp